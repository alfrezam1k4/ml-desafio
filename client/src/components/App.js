import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';

import { Layout } from '.';

const App = () => {
  return (
    <Router>
      <Route render={() => <Layout />} />
    </Router>
  );
};

export default App;
