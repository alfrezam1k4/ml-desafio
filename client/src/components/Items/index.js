import ItemsResult from './ItemsResult';
import Item from './Item';
import ItemDetails from './ItemDetails';

export { ItemsResult, Item, ItemDetails };
