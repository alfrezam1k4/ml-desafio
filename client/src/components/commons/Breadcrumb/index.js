import React from 'react';
import PropTypes from 'prop-types';
import './Breadcrumb.scss';
const Breadcrumb = ({ categories }) => {
  return (
    categories.length > 0 && (
      <p className="breadcrumb">
        {categories.map((crumb, index) => {
          if (index === categories.length - 1)
            return (
              <span key={index}>
                <b> {crumb}</b>
              </span>
            );
          return <span key={index}> {`${crumb} >`} </span>;
        })}
      </p>
    )
  );
};

Breadcrumb.propTypes = {
  categories: PropTypes.array,
};

export default Breadcrumb;
