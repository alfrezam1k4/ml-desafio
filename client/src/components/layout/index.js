import AppBar from './AppBar';
import Layout from './Layout';
import Background from './Background';

export { AppBar, Layout, Background };
