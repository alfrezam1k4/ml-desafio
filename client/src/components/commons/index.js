import { Row, Col } from './Grid';
import Breadcrumb from './Breadcrumb';
export { Row, Col, Breadcrumb };
