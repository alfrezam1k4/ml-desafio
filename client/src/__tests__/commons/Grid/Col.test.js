import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { Row, Col } from '../../../components/commons';

const props = {
  style: { color: 'red' },
  className: 'test-class',
};

const chidren = <div>Test</div>;

describe('<Col />', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Col {...props}>{chidren}</Col>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has correct class', () => {
    const wrapper = mount(<Col {...props}>{chidren}</Col>);
    expect(wrapper.hasClass(props.className)).toBe(true);
  });

  it('has correct styles', () => {
    const wrapper = mount(<Col {...props}>{chidren}</Col>);
    expect(wrapper.get(0).props.style).toHaveProperty('color', 'red');
  });

  test.each([
    ['xs', 6],
    ['sm', 12],
    ['md', 8],
    ['lg', 24],
    ['xl', 16],
    ['xxl', 2],
  ])('should have correct class for size %s', (prop, value) => {
    const sizeProp = { [prop]: value };
    const wrapper = mount(
      <Col {...props} {...sizeProp}>
        {chidren}
      </Col>
    );
    const hasClass = wrapper
      .find('.alf-col')
      .hasClass(`alf-col-${prop}-${value}`);
    expect(hasClass).toBe(true);
  });

  test.each([8, 16, 24, 32, 40, 48])(
    'should have correct col style for row gutter %s',
    (gutter) => {
      const wrapper = mount(
        <Row gutter={gutter}>
          <Col {...props}>{chidren}</Col>
        </Row>
      );
      const styles = wrapper.find('.alf-col').props().style;
      expect(styles).toHaveProperty('paddingLeft', gutter / 2);
      expect(styles).toHaveProperty('paddingRight', gutter / 2);
    }
  );
});
