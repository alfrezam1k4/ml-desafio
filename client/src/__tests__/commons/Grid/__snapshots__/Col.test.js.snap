// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Col /> renders correctly 1`] = `
<div
  className="alf-col test-class"
  style={
    Object {
      "color": "red",
    }
  }
>
  <div>
    Test
  </div>
</div>
`;
